### DISCUSSION TITLE
Cancel a Sequence from within a callback?

### DISCUSSION THREAD

**Discussion by scotsdez** [KyryloKuzyk/PrimeTween#52] [FAQ] [answered]
I have a use case where I have a Sequence that I'd like to repeat an undetermined number of times, but within a callback, I have some logic where I'd like to be able to stop the same Sequence if some conditions are met, is there an easy way to achieve this? If I try to declare the Sequence and use the reference within the same Sequence, the compiler complains that it's unassigned.

```
Sequence autoAdvance = Sequence.Create(cycles: -1, CycleMode.Restart, useUnscaledTime: true)
            .ChainDelay(15f)
            .ChainCallback(() =>
            {
                if (!playerHasManuallyAdvancedText && CurrentPage < AnswerText.textInfo.pageCount)
                {
                    NextPage(false);
                }
                else
                {
                    // cancel this current Sequence (autoAdvance) here
                    // can't reference autoAdvance here as it is undefined
                }
            });```

**Comment by Nelvin** [comment, marked as answer]
This should work

```csharp
Sequence autoAdvance = Sequence.Create(cycles: -1, CycleMode.Restart, useUnscaledTime: true).ChainDelay(15f);
autoAdvance.ChainCallback(() =>
{
    if (!playerHasManuallyAdvancedText && CurrentPage < AnswerText.textInfo.pageCount)
    {
        NextPage(false);
    }
    else
    {
        // cancel this current Sequence (autoAdvance) here
        // can't reference autoAdvance here as it is undefined
        autoAdvance.Stop();
    }
});
```

**Reply by scotsdez** [reply]
Thanks! Makes perfect sense, not sure why I didn't think to separate out the assignment and then use the reference on the next line.

**Reply by KyryloKuzyk** [reply]
Because in C# the variable must be initialised before the first usage. The right side of assignment may theoretically throw an exception, so C# can't guarantee that the variable will be initialised. 
Huge thanks @Nelvin for answering this!
